### DISCUSSION TITLE
Will Rocketmq be maintained after 4.9? Do we need to immediately switch to 5.0？

### DISCUSSION THREAD

**Discussion by luozongle** [apache/rocketmq#6664] [General]
Will Rocketmq be maintained after 4.9? Do we need to immediately switch to 5.0？

**Comment by RongtongJin** [comment]
> Will Rocketmq be maintained after 4.9? Do we need to immediately switch to 5.0？

Before January 2024, RocketMQ 4.9.x will continue to be maintained, but only for bug fixes and will not add new features. Please refer to https://rocketmq.apache.org/download. If you need some new features of RocketMQ 5.0, such as proxy, arbitrarily precise timing messages, automatic master-slave switching architecture, it is recommended that you upgrade to version 5.0.

**Comment by mxsm** [comment]
@luozongle 
4.9.x will not introduce new features, only bug fixes and some performance improvements. If you want to use new features such as:

1. Delayed messages at any time
2. Proxy mode
3. New high availability Controller mode

And other new features, you need to upgrade to Rocketmq 5.0 or above. Of course, the basic functions of version 5.0 and 4.x are compatible. If you have new requirements or use peripheral ecosystem projects of Rocketmq, it is recommended to upgrade to 5.0. Some functions of the 5.0 upgrade process may not be compatible with version 4.x. Please refer to the official website documentation for specific upgrade steps.
